dashboard/test/controllers/followers_controller_test.rb

```diff
@@ -69,6 +69,42 @@ class FollowersControllerTest < ActionController::TestCase
     assert_template 'followers/student_user_new'
   end
 
+  test "student_user_new increments join section attempts when signed in" do
+    refute @chris_section.students.include? @student
+    sign_in @student
+    section_join_attempts = @student.num_failed_section_attempts || 0
+    assert_does_not_create(User) do
+      post :student_register, params: {section_code: @chris_section.code}
+    end
+    @student.reload
+    updated_section_join_attempts = @student.num_failed_section_attempts
+    change_in_section_attempts = updated_section_join_attempts - section_join_attempts
+    assert_equal 1, change_in_section_attempts
+  end
+
+  test 'student_user_new displays captcha when joining 3 or more sections in 24 hours' do
+    @new_student = create(:user)
+    sign_in @new_student
+    3.times do
+      post :student_register, params: {section_code: 'INVALID'}
+      @new_student.reload
+    end
+    assert_equal true, @new_student.display_captcha?
+  end
+
+  # Enable reCAPTCHA gem in test env for just this unit test.
+  # Restores original configuration after testing the assertion
+  test 'student_user_new displays error when joining 3 or more sections in 24 hours without completing captcha' do
+    Recaptcha.configuration.skip_verify_env.delete("test")
```

Comment:
I poked around a bit and found this pattern for stubbing calls to an external library/service.  Maybe you can try stubbing out the call to verify_recaptcha entirely?  (If that doesn't work, what you have here is ok too.)

Reply:
@jamescodeorg Did you forget to paste a link here?

Reply:
Ooops, here's the link: https://github.com/code-dot-org/code-dot-org/blob/f970daeba23a8630f5d470fe18349753befeabc2/lib/test/cdo/aws/s3.rb#L10.  Maybe you can stub verify_recaptcha the way list_objects_v2 is stubbed in that test.